File: week11/20/pr2_min_cost_climbing_stairs.c
int min(int a, int b){
    return a <= b ? a : b;
}
// 아래에서 위로
int minCostClimbingStairs(int* cost, int costSize) {
    int dp[costSize];
    dp[0] = cost[0];
    dp[1] = cost[1];
    for(int i=2;i<costSize;i++){
        dp[i] = min(dp[i-1],dp[i-2]) + cost[i];
    }
    return min(dp[costSize-1],dp[costSize-2]);
}

File: week11/20/pr2_min_cost_climbing_stairs.py
# temp = [0]*10
# print(temp)
# print(temp[2])
# test = [3,4,5,6]
# for i in range(3,-1,-1):
#     print(test[i])
class Solution:
    def minCostClimbingStairs(self, cost: list[int]) -> int:
        dp = [0]*(len(cost)+2)
        for i in range(len(cost)-1,-1,-1):
            dp[i] = cost[i] + min(dp[i+1],dp[i+2])
        return min(dp[0],dp[1])